# _Branchiostoma lanceolatum_ from Marlétaz et al. (2018)

Note: The genome here is not the assembled and post-processed North Sea specimen of _B. lanceolatum_ but rather a previously published genome from [Marlétaz et al. (2018)](https://www.nature.com/articles/s41586-018-0734-6).

## BRAKER ET mode

### Align RNA-seq reads using Hisat2

Remove white splaces in the fasta header

```bash
sed 's, ,_,g'<Branchiostoma_lanceolatum.BraLan2.dna_sm.toplevel.fa >Blnc2018.fa
```

Where `Branchiostoma_lanceolatum.BraLan2.dna_sm.toplevel.fa` is the downloaded draft genome.

Index the draft genome

```bash
hisat2-build \
../Blnc2018.fa \
Blnc2018
```
Where `Blnc2018.fa` is the scaffolded assembly (repeat-masked, haplotig-purged and polished).

Map the RNA-seq reads

```bash
hisat2 \
-x Blnc2018 \
-1 ../../../Embryo_RNA_preprocessing/Blnc_all_RNA_R1_trimmed.fq \
-2 ../../../Embryo_RNA_preprocessing/Blnc_all_RNA_R2_trimmed.fq \
-p 10 \
-S input.sam
```

Where `Blnc_all_RNA_R1_trimmed.fq` and `Blnc_all_RNA_R2_trimmed.fq` are the forward (R1) and reverse (R2) trimmed RNA-seq reads from both adult and embryo. The embryo reads were trimmed using trimmomatic, i.e.

```bash
trimmomatic PE \
-threads 10 \
-trimlog fastqc2.log \
Blnc_EMBRYO_RNA_R1_raw.fastq \
Blnc_EMBRYO_RNA_R2_raw.fastq \
Blnc_EMBRYO_RNA_R1_trim_2.fq.gz \
Blnc_EMBRYO_RNA_R1_unpaired_2.fq.gz \
Blnc_EMBRYO_RNA_R2_trim_2.fq.gz \
Blnc_EMBRYO_RNA_R2_unpaired_2.fq.gz \
ILLUMINACLIP:adapter.fa:4:30:10 \
SLIDINGWINDOW:4:15 \
MINLEN:36 \
LEADING:3 \
TRAILING:3
```

The output was unzipped, i.e.

```bash
gunzip -c Blnc_EMBRYO_RNA_R1_trim_2.fq.gz >Blnc_EMBRYO_RNA_R1_trim_2.fq
```

```bash
gunzip -c Blnc_EMBRYO_RNA_R2_trim_2.fq.gz >Blnc_EMBRYO_RNA_R2_trim_2.fq
```
...and concatenated with the adult reads.

```bash
cat Blnc_EMBRYO_RNA_R1_trim_2.fq Blnc_ADULT_RNA_R1_trim.fq > Blnc_all_RNA_R1_trimmed.fq
```

```bash
cat Blnc_EMBRYO_RNA_R2_trim_2.fq Blnc_ADULT_RNA_R2_trim.fq > Blnc_all_RNA_R2_trimmed.fq
```

Where `Blnc_ADULT_RNA_R1_trim.fq` and `Blnc_ADULT_RNA_R2_trim.fq` are the adult reads.

Convert `.sam` to `.bam`

```bash
samtools view -bS input.sam > input.bam
```
Sort

```bash
samtools sort -n -@ 4 -m 2G input.bam -o input.sorted.bam
```

### BRAKER ET mode (BRAKER v2.1.6)

```bash
braker.pl \
--species=Blnc2018_ET_adult_embryo_3 \
--genome=../Blnc2018.fa \
--bam=input.sorted.bam \
--softmasking \
--cores=20
```

Where `input.sorted.bam` is the RNA-seq read alignment file and `Blnc2018.fa` is the scaffolded assembly.

## Protein sequence extraction

### Get CDS using getAnnoFasta.pl (BRAKER v2.1.6)

```bash
getAnnoFasta.pl ../braker/braker.gtf --seqfile ../../Blnc2018.fa
```

Where `Blnc2018.fa` is the published draft genome and `braker.gtf` is the BRAKER annotation file.

### Use transeq from EMBOSS to translate CDS

```bash
transeq ../braker/braker.codingseq Blnc2018_ET_all.faa
```

Where `braker.codingseq` is the output of the previous step.
